x-pack/plugins/enterprise_search/server/lib/analytics/delete_analytics_collection.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from '@kbn/core-elasticsearch-server';
+
+import { ANALYTICS_COLLECTIONS_INDEX } from '../..';
+
+import { ErrorCode } from '../../../common/types/error_codes';
+import { fetchIndices } from '../indices/fetch_indices';
+
+import { fetchAnalyticsCollectionByName } from './fetch_analytics_collection';
+
+const deleteAnalyticsCollectionEvents = async (client: IScopedClusterClient, name: string) => {
+  const indexPattern = `elastic_analytics-events-${name}-*`;
+  const indices = await fetchIndices(client, indexPattern, true, false);
+
+  for (const index of indices) {
+    await client.asCurrentUser.indices.delete({
+      ignore_unavailable: true,
+      index: index.name,
```

Comment:
suggestion: You should be able pass all the index names to this property as an array of index names, instead of doing N single-index calls in a loop.

Reply:
oooh interesting. Thanks, will try that out.

Reply:
updated! Tested and works still as expected.